core/src/main/java/io/kestra/core/plugins/PluginResolver.java

```diff
@@ -35,7 +39,32 @@ private static boolean isClassFile(final Path path) {
         return path.toString().toLowerCase().endsWith(".class");
     }
 
-    public List<ExternalPlugin> resolves() {
+    public void watch(Consumer<ExternalPlugin> onNewPlugin) throws Exception {
+        this.initialPlugins().forEach(onNewPlugin);
+        watchService = FileSystems.getDefault().newWatchService();
+        pluginPath.register(watchService, StandardWatchEventKinds.ENTRY_CREATE, StandardWatchEventKinds.ENTRY_MODIFY);
+        Set<Path> createdWaitingForWrite = new HashSet<>();
+        watcherThread.execute(throwRunnable(() -> {
+            WatchKey key;
+            try {
+                while ((key = watchService.take()) != null) {
+                    for (WatchEvent<?> event : key.pollEvents()) {
+                        Path eventPath = pluginPath.resolve((Path) event.context());
+                        if (event.kind() == StandardWatchEventKinds.ENTRY_CREATE) {
+                            createdWaitingForWrite.add(eventPath);
+                        } else if (event.kind() == StandardWatchEventKinds.ENTRY_MODIFY && createdWaitingForWrite.remove(eventPath)) {
+                            onNewPlugin.accept(toExternalPlugin(eventPath));
+                        }
```

Comment:
As there is only one thread I don't think it's needed to check that an entry is concurrently create/modify.
Moreover it seems that on CREATE it is not added?

Or maybe we have two events when we create a file? First create then modify? In this case we can only listen or modify.

Reply:
There is duplicates in Modify event that's why I had to put a "CREATE then MODIFY" listener.
Create event is triggered when the resource has not its content yet so it cannot be used on its own

Reply:
OK, understood.